app/earn/[...vault]/hooks/use-vault-context.tsx

```diff
@@ -0,0 +1,62 @@
+import type { PropsWithChildren } from 'react'
+import { createContext, useContext, useState } from 'react'
+import { useNumberInputValue } from './use-number-input-value'
+import type { Vault } from '@/api'
+import { FormAction } from '@/store/slices/vaultActionSlice'
+
+interface VaultContextType {
+  inputValue: bigint
+  displayValue: string
+  formAction: FormAction
+  insured: boolean
+  vault: Vault
+  onValueChange: (value: string | bigint) => void
+  setFormAction: (action: FormAction) => void
+  setInsured: (insured: boolean) => void
+}
+
+let VaultContext: React.Context<VaultContextType>
+
+function createVaultContext(vault: Vault) {
+  if (VaultContext)
+    return VaultContext
+
+  return (VaultContext = createContext<VaultContextType>({
+    inputValue: 0n,
+    displayValue: '0',
+    formAction: FormAction.DEPOSIT,
+    insured: true,
+    vault,
+    onValueChange: () => {},
+    setFormAction: () => {},
+    setInsured: () => {},
+  }))
+}
```

Comment:
Why it creates context like a static method creates singltone? Context in general singltone, I think it must be created in advance, not durign runtime. Better to expect that `Vault` insude can be empty and show some loader when it not, rather than break type system to think that it allways set